_posts/elements/2015-05-12-forms.md

```diff
@@ -0,0 +1,284 @@
+---
+layout:       element
+title:        Forms
+date:         2015-05-12 16:02:35
+
+categories:   elements
+tags:
+- atomic
+- pattern
+- forms
+
+slug:         forms
+scssurl:      "https://github.com/edx/ux-pattern-library/blob/master/_src/edx-pl/sass/components/_forms.scss"
+
+description:  Various examples of different form fields and their usage and options.
+
+info: Forms allow users to interact with the interface, set preferences, and enter any necessary personal information. They also form the base of individual exercises within each class.
+---
+
+<h3 class="hd-4 example-set-hd">Forms</h3>
+<h4 class="hd-6">Text inputs</h4>
+<div class="example-set">
+    <fieldset class="fieldset">
+        <legend class="group-legend is-hidden">Personal Details</legend>
+        <div class="field-wrapper">
+            <label class="form-label" for="full_name_default">Your name (default text field)</label>
+            <input class="input input-text" type="text" name="full_name_default" id="full_name_default" placeholder="e.g. Jane Doe" required>
+        </div>
+        <div class="field-wrapper">
+            <label class="form-label" for="email_address_default">Email address (default, disabled text field)</label>
+            <input class="input input-text" type="email" name="email_address_default" id="email_address_default" placeholder="e.g. jane@email.com" disabled>
+        </div>
+        <div class="field-wrapper">
+            <label class="form-label required-field" for="email_address_default_helper">Email address (default, required text field with helper text)</label>
```

Comment:
It should be on the label.

The input/field itself is designated `required` via an attribute, but we need to tell the `label`'s that its field is required, so we put this on the `label`. That cool?